app/routers/profile.py

```diff
@@ -153,3 +168,71 @@ def get_image_crop_area(width, height):
         return delta, 0, width - delta, height
     delta = (height - width) // 2
     return 0, delta, width, width + delta
+
+
+@router.post("/holidays/update")
+async def update_holidays(
+        file: UploadFile = File(...), session=Depends(get_db)):
+    icsfile = await file.read()
+    holidays = get_holidays_from_file(icsfile.decode(), session)
+    try:
+        save_holidays_to_db(holidays, session)
+    except SQLAlchemyError as ex:
+        logger.exception(ex)
+    finally:
+        url = router.url_path_for("profile")
+        return RedirectResponse(url=url, status_code=HTTP_302_FOUND)
+
+
+def get_holidays_from_file(file: List[Event], session: Session) -> List[Event]:
```

Comment:
I think these functions should be in the `internal` directory. WDYT?

Reply:
i think you right. but it is related to my second feature so ill fix it there.